recyclerview/src/main/java/me/henrytao/recyclerview/BaseAdapter.java

```diff
@@ -274,15 +296,31 @@ public void onItemRangeRemoved(int positionStart, int itemCount) {
           }
         }
       };
-      mBaseAdapter.registerAdapterDataObserver(mAdapterDataObserver);
+      for (RecyclerView.Adapter adapter : mBaseAdapters) {
+        adapter.registerAdapterDataObserver(mAdapterDataObserver);
+      }
     }
     if (notifyDataSetChanged) {
       notifyDataSetChanged();
     }
   }
 
+  private int[] getBaseAdapterInfo(int dataPosition) {
```

Comment:
Are there any ways to make this method explicit? Please use `protected` and rename it to `getBaseAdapter`. It should return `baseAdapterIndex`, `dataPosition` and `adapter itself`.